packages/aws-cdk/test/integ/cli/resource-pool.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * A class that holds a pool of resources and gives them out and returns them on-demand
+ *
+ * The resources will be given out front to back, when they are returned
+ * the most recently returned version will be given out again (for best
+ * cache coherency).
+ *
+ * If there are multiple consumers waiting for a resource, consumers are serviced
+ * in FIFO order for most fairness.
+ */
+export class ResourcePool<A> {
```

Comment:
Unit tests for the leasing logic.  A regression here (now or in the future) would be hard to catch/debug, esp one that works partially causing the logic to skew but not fail.